src/cfg/cfg-traversal.h

```diff
@@ -69,10 +69,19 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
   std::vector<BasicBlock*> ifStack;
   // stack of the first blocks of loops
   std::vector<BasicBlock*> loopStack;
+
   // stack of the last blocks of try bodies
   std::vector<BasicBlock*> tryStack;
-  // stack of the first blocks of catch bodies
-  std::vector<BasicBlock*> catchStack;
+  // stack of the first blocks of catches that throwing instructions should
+  // unwind to at any moment. Because there can be multiple catch blocks for a
+  // try, we maintain a vector of first blocks of catches.
+  std::vector<std::vector<BasicBlock*>> unwindCatchStack;
+  // stack of 'Try' expression corresponding to unwindCatchStack.
+  std::vector<Expression*> unwindExprStack;
+  // Contains the stack of the first blocks of catches currently being processed
+  // in the beginning. Will contain the last blocks of catches after being
+  // processed.
```

Comment:
this comment is a little unclear to me. What does "beginning" mean?

Reply:
I tried to explain this in https://github.com/WebAssembly/binaryen/pull/3494#discussion_r558528017. So this vector stores catch start BBs in `doStartTry` so that each `doStartCatch` can retrieve its start BB. After processing each catch `doEndCatch` stores each catch's end BB so that `doEndTry` can access it.

Reply:
If we decide to keep a single vector for this, how about
```
// A stack for each try, where each entry is a list of blocks, one for
// each catch, used during processing. We start by assigning the
// start blocks to here, and then read those at the appropriate time;
// when we finish a catch we write to here the end block, so that when
// we finish with them all we can connect the ends to the outside. In
// principle two vectors could be used, but their usage does not overlap
// in time, and this is more efficient.
```

Reply:
Thanks! Done.